src/protocol/boolean/xor.rs

```diff
@@ -0,0 +1,74 @@
+use crate::error::BoxError;
+use crate::ff::Field;
+use crate::protocol::{context::ProtocolContext, mul::SecureMul, RecordId};
+use crate::secret_sharing::Replicated;
+
+/// Secure XOR protocol with two inputs, `a, b ∈ {0,1} ⊆ F_p`.
+/// It computes `[a] + [b] - 2[ab]`
+pub struct Xor<F: Field> {
+    a: Replicated<F>,
+    b: Replicated<F>,
+}
+
+impl<'a, F: Field> Xor<F> {
+    #[allow(dead_code)]
+    pub fn new(a: Replicated<F>, b: Replicated<F>) -> Self {
+        Self { a, b }
+    }
```

Comment:
As mentioned above, I see no gain to instantiating this struct. Let's just make it a funtion.